xls/public/c_api_dslx.cc

```diff
@@ -167,6 +167,15 @@ struct xls_dslx_enum_def* xls_dslx_module_get_type_definition_as_enum_def(
   return reinterpret_cast<xls_dslx_enum_def*>(cpp_enum_def);
 }
 
+struct xls_dslx_type_alias* xls_dslx_module_get_type_definition_as_type_alias(
+    struct xls_dslx_module* module, int64_t i) {
+  auto* cpp_module = reinterpret_cast<xls::dslx::Module*>(module);
+  xls::dslx::TypeDefinition cpp_type_definition =
+      cpp_module->GetTypeDefinitions().at(i);
+  auto* cpp_type_alias = std::get<xls::dslx::TypeAlias*>(cpp_type_definition);
```

Comment:
are we ok `std::bad_variant_access` here, or should we catch and return `nullptr`?

Reply:
Similar functions just state it's a precondition so I did similar here -- note that the library is built with -fno-exceptions so it should std::terminate on exception.

Reply:
but is that different from `xls_dslx_type_definition_get_colon_ref` below which does check the return value from `std::get`? (is the difference that the later works directly w/ the `type_definition` object so that we expect people might want to check different scenario while this assume intent, i.e: the caller knows that the type definition at `i` is a `type_alias`?

Reply:
Ah, this is because TypeDefinition is literally a variant. This allows us to perform both std::holds_alternative and std::get in a single call, it's like std::get_if. Pretty much all other functions here are not just reading out from a variant where we don't know the contents.

I added a comment in the `.h` on this.